File: solutions/cpp/hamming/2/hamming.cpp
#include "hamming.h"
#include <stdexcept>
#include <numeric>
namespace hamming {
    std::size_t compute(std::string_view first, std::string_view second) {
        if(first.size() != second.size())
            throw std::domain_error("");
        return std::inner_product(first.begin(), first.end(), second.begin(), 0, std::plus<>(), std::not_equal_to<>());
    }

}  // namespace hamming

File: solutions/lua/hamming/2/hamming.lua
local Hamming = {}

function Hamming.compute(a,b)
    if #a ~= #b then    
        return -1
    end
    
    local diffCount = 0
    for i = 1, #a do
        if a:byte(i) ~= b:byte(i) then
            diffCount = diffCount + 1
        end
    end
    return diffCount
end

return Hamming
